syntax/std_seq_array_helper.go

```diff
@@ -0,0 +1,206 @@
+package syntax
+
+import (
+	"github.com/arr-ai/arrai/rel"
+)
+
+// arrayContains check if array a contains b, and b can be rel.Value or rel.Array.
+func arrayContains(a rel.Array, b rel.Value) rel.Value {
+	bArray := convert2Array(b)
+	return rel.NewBool(indexSubArray(a.Values(), bArray.Values()) > -1)
+}
+
+// arraySub substitutes all old in a with new.
+func arraySub(a rel.Array, old, new rel.Value) rel.Value {
+	// Convert to array to facilitate process
+	oldArray := convert2Array(old)
+	newArray := convert2Array(new)
+
+	finalVals := make([]rel.Value, 0, a.Count())
+	for start, absoluteIndex := 0, 0; start < a.Count(); {
+		relativeIndex := indexSubArray(a.Values()[start:], oldArray.Values())
+		if relativeIndex >= 0 {
+			absoluteIndex = relativeIndex + start
+			if absoluteIndex-start > 0 {
+				finalVals = append(finalVals, a.Values()[start:absoluteIndex]...)
+			}
+			finalVals = append(finalVals, newArray.Values()...)
+			start = absoluteIndex + oldArray.Count()
+		} else {
+			finalVals = append(finalVals, a.Values()[absoluteIndex+1:]...)
+			break
+		}
+	}
+
+	return rel.NewArray(finalVals...)
+}
+
+// arraySplit split a by b.
+func arraySplit(a rel.Array, b rel.Value) rel.Value {
+	delimiter := convert2Array(b)
+	var result []rel.Value
+
+	if delimiter.Count() == 0 {
+		for _, e := range a.Values() {
+			result = append(result, rel.NewArray(e))
+		}
+	} else {
+		for start, absoluteIndex := 0, 0; start < a.Count(); {
+			relativeIndex := indexSubArray(a.Values()[start:], delimiter.Values())
+			if relativeIndex >= 0 {
+				absoluteIndex = relativeIndex + start
+				if start != absoluteIndex {
+					result = append(result, rel.NewArray(a.Values()[start:absoluteIndex]...))
+				}
+				start = absoluteIndex + delimiter.Count()
+			} else {
+				if start == 0 || start < a.Count() {
+					result = append(result, rel.NewArray(a.Values()[start:]...))
+				}
+				break
+			}
+		}
+	}
+
+	return rel.NewArray(result...)
+}
+
+// arrayJoin joins array b to a, b is joiner and a is joinee.
+func arrayJoin(a rel.Array, b rel.Value) rel.Value {
+	joiner := convert2Array(b)
+	if joiner.Count() == 0 || a.Count() == 0 {
+		// if joinee is empty, the final value will be empty
+		return a
+	}
+
+	vals := make([]rel.Value, 0, a.Count())
+	for i, value := range a.Values() {
+		switch vArray := value.(type) {
+		case rel.Array:
+			vals = append(vals, generate1LevelArray(vArray)...)
+		case rel.Value:
+			vals = append(vals, value)
+		}
+
+		if i+1 < a.Count() {
```

Comment:
Move this above the switch and change it to:
```suggestion
		if i > 0 {
```

Reply:
Fixed.